plugins/lookup/secretsmanager_random_password.py

```diff
@@ -0,0 +1,191 @@
+            )
+
+        on_denied = self.get_option("on_denied")
+
+        # validate arguments 'on_denied'
+        if on_denied is not None and (
+            not isinstance(on_denied, string_types) or on_denied.lower() not in ["error", "warn", "skip"]
+        ):
+            raise AnsibleLookupError(
+                f'"on_denied" must be a string and one of "error", "warn" or "skip", not {on_denied}'
+            )
+
+        params = {}
+        # validate password length argument or option
+        self.debug(f"Options: {self.get_options()}")
+        password_length = self.get_option("password_length")
+        if len(terms) == 1:
+            if password_length is not None:
+                raise AnsibleLookupError(
+                    f'"password_length" should be provided as argument or keyword, not both'
+                )
+            password_length = terms[0]
+        if password_length is not None:
+            if not isinstance(password_length, integer_types) or password_length < 1:
+                raise AnsibleLookupError(
+                    f'"password_length" must be an integer greater than zero, if provided'
+                )
+            params["PasswordLength"] = password_length
+
+        # validate exclude characters
+        exclude_characters = self.get_option("exclude_characters")
+        if exclude_characters is not None:
+            if not isinstance(exclude_characters, string_types):
+                raise AnsibleLookupError(
+                    f'"exclude_characters" must be a string, if provided'
+                )
+            params["ExcludeCharacters"] = exclude_characters
+
+        # validate options for parameters
+        bool_options_to_params = {
+            "exclude_numbers": "ExcludeNumbers",
+            "exclude_punctuation": "ExcludePunctuation",
+            "exclude_uppercase": "ExcludeUppercase",
+            "exclude_lowercase": "ExcludeLowercase",
+            "include_space": "IncludeSpace",
+            "require_each_included_type": "RequireEachIncludedType",
+        }
+        for opt_name in bool_options_to_params.keys():
+            opt_value = self.get_option(opt_name)
+            if opt_value is not None:
+                if not isinstance(opt_value, bool):
+                    raise AnsibleLookupError(
+                        f'"{opt_name}" must be a boolean value, if provided'
+                    )
+                params[bool_options_to_params[opt_name]] = opt_value
+
+        client = self.client("secretsmanager", AWSRetry.jittered_backoff())
+
+        try:
+            response = client.get_random_password(**params)
+            return [response['RandomPassword']]
+        except is_boto3_error_code("AccessDeniedException"):  # pylint: disable=duplicate-except
+            if on_denied == "error":
+                raise AnsibleLookupError(f"Failed to generate random password (AccessDenied)")
```

Comment:
We need the origin AccessDenied message, otherwise it's unclear which permissions are missing. Because `secretsmanager:Get*` is already granted.